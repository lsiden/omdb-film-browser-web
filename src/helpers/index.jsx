export function queryFromUriPath(path) {
  const match = /\/search\/([^/?]+)/.exec(decodeURI(path))
  return match ? match[1] : ''
}
