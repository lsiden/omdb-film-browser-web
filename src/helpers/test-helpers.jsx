import React from 'react'
import { Provider } from 'react-redux'
import { StaticRouter } from 'react-router-dom'

/* eslint-disable-next-line import/no-extraneous-dependencies */
import { create } from 'react-test-renderer'

import store from 'store'

export const createWithContext = (ui) => create(
  <Provider store={store}>
    <StaticRouter context={{}}>
      {ui}
    </StaticRouter>
  </Provider>
)
