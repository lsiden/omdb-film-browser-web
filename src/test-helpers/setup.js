import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

jest.mock("abort-controller", () => class {})
jest.mock("event-target-shim", () => {})
