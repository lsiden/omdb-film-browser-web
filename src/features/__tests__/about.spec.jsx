import React from 'react'

import About from 'features/about'
import { createWithContext } from 'helpers/test-helpers'

const onClick = () => {}
const createWrapper = () => createWithContext(<About onClick={onClick} />)

describe('OmdbMain', () => {
  test('matches snapshot', () => {
    expect(createWrapper()).toMatchSnapshot()
  })
})
