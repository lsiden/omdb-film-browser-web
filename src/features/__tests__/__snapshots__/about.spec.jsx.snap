// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OmdbMain matches snapshot 1`] = `
<div
  style={
    Object {
      "outline": "none",
    }
  }
>
  <header
    style={
      Object {
        "background": "linear-gradient(to right, #000040, #3b00ff)",
        "color": "white",
        "padding": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <h1>
        <img
          height={20}
          src="movie-icon.svg"
          style={
            Object {
              "margin": 12,
            }
          }
        />
         
        Open Movie Database Browser
      </h1>
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
      >
        About
      </a>
    </div>
  </header>
  <header
    style={
      Object {
        "background": "linear-gradient(to right, #000040, #3b00ff)",
        "color": "white",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "10px",
      }
    }
  >
    <h1
      style={
        Object {
          "fontSize": 28,
        }
      }
    >
      About Open Movie Database Browser
    </h1>
    <button
      className="btn btn-link"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
      title="Close"
      type="button"
    >
      âœ•
    </button>
  </header>
  <div
    style={
      Object {
        "marginLeft": 20,
      }
    }
  >
    <div>
      <h2>
        Written By
      </h2>
      <p>
        Lawrence Siden
      </p>
      <p>
        <a
          href="http://westsideconsultingllc.com"
        >
          Westside Consulting LLC
        </a>
      </p>
      <p>
        <a
          href="mailto:westsideconsultingllc@gmail.com"
        >
          westsideconsultingllc@gmail.com
        </a>
      </p>
      <h2>
        Credits
      </h2>
      <p>
        <a
          href="https://www.omdbapi.com/"
        >
          Open Movie Database API
        </a>
      </p>
      <h2>
        Source Code
      </h2>
      <p>
        <a
          href="https://github.com/lsiden/omdb-film-browser-web"
        >
          https://github.com/lsiden/omdb-film-browser-web
        </a>
      </p>
    </div>
  </div>
</div>
`;
